core/lib/rom/plugin_registry.rb

```diff
@@ -61,48 +27,48 @@ def initialize
     # @option options [Symbol] :adapter (:default) which adapter this plugin
     # applies to. Leave blank for all adapters
     def register(name, mod, options = EMPTY_HASH)
-      type    = options.fetch(:type)
-      adapter = options.fetch(:adapter, :default)
+      type(options.fetch(:type)).register(name, mod, options)
+    end
 
-      plugins_for(type, adapter).register(name, mod, options)
+    # @api private
+    def type(type)
+      types.fetch_or_store(type) do
+        if Plugins[type][:adapter]
+          AdapterPluginsContainer.new(type)
+        else
+          PluginsContainer.new({}, type: type)
+        end
+      end
     end
 
-    private
+    # @api private
+    def [](type)
+      types.fetch(singularize(type))
+    end
 
-    # Determine which specific registry to use
+    # Old API compatibility
     #
     # @api private
-    def plugins_for(type, adapter)
+    def singularize(type)
       case type
-      when :configuration then configuration
-      when :command       then commands.adapter(adapter)
-      when :mapper        then mappers.adapter(adapter)
-      when :relation      then relations.adapter(adapter)
-      when :schema        then schemas.adapter(adapter)
+      when :relations then :relation
+      when :commands then :command
+      when :mappers then :mapper
+      when :schemas then :schema
+      else type
       end
```

Comment:
@flash-gordon should we maybe use the inflector here?

Reply:
I thought about it but I see no point, better remove these aliases in the next version.